### ISSUE TITLE
[Errno 10035] A non-blocking socket operation could not be completed immediately

### ISSUE THREAD

**Issue by ivankravets** [pyserial/pyserial#235]
The latest v3.3 release has a bug with opening serial monitor via socket port. 
v3.2.1 works very well!

```
could not open port ‘socket://localhost:64256’: reset_input_buffer failed: [Errno 10035] A non-blocking socket operation could not be completed immediately
```

**Comment by ivankravets** [comment]
Thanks, fixed in 3.4!

**Comment by jasvi** [comment]
Yay! I was struggling to figure out what I was doing wrong. Thanks.
